content/courses/onchain-development/local-setup.md

```diff
@@ -81,32 +105,35 @@ anchor test
 ```
 
 **The `anchor test` command should complete with no errors or warnings**.
-However you may encounter issues, and we'll fix them below:
 
-##### `package `solana-program v1.18.12` cannot be built because it requires rustc 1.75.0 or newer` error
+**However you may encounter issues, and we'll fix them below:**
+
+- `package `solana-program
+  v1.18.12` cannot be built because it requires rustc 1.75.0 or newer` error
 
+This error is due to incompatible versions of `solana-program` and `solana-cli`.
 Run `cargo add solana-program@"=1.18.x"`, where `x` matches your version of
 `solana-cli`. Then re-run `anchor test`.
 
-##### `Error: Unable to read keypair file`
+- `Error: Unable to read keypair file`
 
 Add a keypair to `.config/solana/id.json`. You can either copy a keypair from an
 `.env` file (just the array of numbers) into a file or use the command
 `solana-keygen new --no-bip39-passphrase` to create a new keypair file. Then
 re-run `anchor test`.
 
-##### `unused variable: 'ctx'` warning
+- `unused variable: 'ctx'` warning
 
 This simply means the `initialize` instruction handler isn't doing anything yet.
 You can open `programs/favorites/src/lib.rs` and change `ctx` to `_ctx` or just
 go onto the next step.
 
-##### `No license field in package.json` warning
+- `No license field in package.json` warning
```

Comment:
A couple of other ones to add:

```
#### error: no such command: `build-sbf`

export PATH=~/.local/share/solana/install/active_release/bin:$PATH  Add this to your ~/.zshrc or ~/.bashrc file to make the change permanent.

### Unable to get latest blockhash. Test validator does not look started. 

Install Homebrew and use it to install GNU tar:  brew install gnu-tar
export PATH=/opt/homebrew/opt/gnu-tar/libexec/gnubin:$PATH Add this to your ~/.zshrc or ~/.bashrc file to make the change permanent.


```

(please format nicely per above)

Reply:
got it

Reply:
should I add some explanation - like their cause - for the errors listed in the doc?